File: AtCoder Beginners Selection/Shift only.rb
n = gets.to_i
nums = gets.split.map(&:to_i)
count = 1
nums.each do |x|
    if x%2 == 1
        puts 0
        exit
    end
end

loop do 
    i=0
    nums.each do |x|
        nums[i]=x/2
        if(nums[i]%2==1)
            puts count
            exit
        end
        i+=1
    end
    count+=1
end

File: AtCoder Beginners Selection/Shift only.cpp
#include<iostream>
#include<string>
using namespace std;

int main(){
    int count=0;
    int n;
    unsigned long long int  nums[201];
    bool even = true;
    cin >> n;
    for(int i = 0;i < n;i++)
    {
        cin >> nums[i];
        if(nums[i]%2==1)
            even = false;
    }
    while (even){  
        for(int j = 0;j < n;j++){
            nums[j]=nums[j]/2;
            if(nums[j]%2==1){
                even = false;
                break;
            }
        }
        count+=1;
    }
    cout << count << endl;
    return 0;
}